frontend/exporter/src/constant_utils.rs

```diff
@@ -255,6 +259,52 @@ mod rustc {
                 });
                 ConstantExprKind::Literal(scalar_int_to_constant_literal(s, scalar_int, ty))
             }
+            ty::Float(float_type) => {
+                use rustc_apfloat::Float;
+                let value: u128 = match float_type {
+                    ty::FloatTy::F16 => {
+                        let f = scalar.to_f16().unwrap_or_else(|_| {
+                            fatal!(
+                                s[span],
+                                "Type is primitive, but the scalar {:#?} is not a [Float]",
+                                scalar
+                            )
+                        });
+                        rustc_apfloat::ieee::Half::to_bits(f)
+                    }
```

Comment:
You can go directly from the scalar to its bits without round-tripping through a structured representation: `scalar.assert_scalar_int().to_bits_unchecked()`

Reply:
Thanks! Modified in https://github.com/hacspec/hax/pull/880/commits/005594778f2aff4111e4f47fd893f52e29f84263, I went through `try_to_scalar_int` to provide a better error message